### PR TITLE
Host cleanup

### PR DESCRIPTION
**Pull request by ankilosaurus**
Problem:
- hosts deleted from outside of Teletraan (ex: deleted using aws console or cli etc.) are not cleaned up from Teletraan hosts/agents table. (If host is terminated from teletraan using Terminate/Force Terminate option works fine)
Current query - "SELECT DISTINCT hosts.* FROM hosts INNER JOIN hosts_and_envs ON hosts.host_name=hosts_and_envs.host_name WHERE hosts.last_update<?" - does not cleanup all the stale hosts.

Due to this hosts and agents table keeps growing with those dead hosts. 

Fix:
Remove stale hosts/agents based on last update timestamp from hosts and agents tables.

Test pending.

### CODE REVIEW

File: deploy-service/common/src/main/java/com/pinterest/deployservice/handler/GoalAnalyst.java [hunk 1]
```diff
@@ -162,9 +162,20 @@ public String toString() {
...
+            if (systemPriority != null) {
+                return systemPriority;
+            }
+            return environ.getPriority().getValue();
+        }
+
         @Override
         public int compareTo(UninstallCandidate uninstallCandidate) {
-            return 0;
+            int priority1 = getUninstallPriority(environ);
```
Comment (ankilosaurus) [inline deploy-service/common/src/main/java/com/pinterest/deployservice/handler/GoalAnalyst.java:175]: not part of the change. will revert

File: deploy-service/teletraanservice/src/main/java/com/pinterest/teletraan/worker/SimpleAgentJanitor.java [hunk 2]
```diff
@@ -43,30 +42,25 @@
...
     // remove the stale host from db
     void removeStaleHost(String id) throws Exception {
-        try {
-            hostDAO.deleteAllById(id);
-            hostAgentDAO.delete(id);
-            LOG.info("AgentJanitor delete all records for host {}.", id);
-        } catch (Exception e) {
-            LOG.error("AgentJanitor Failed to delete all records for host {}", id, e);
-        }
+        LOG.info(String.format("Delete records of stale host {}", id));
```
Comment (disparter) [inline deploy-service/teletraanservice/src/main/java/com/pinterest/teletraan/worker/SimpleAgentJanitor.java:62]: Without this try catch the exception will be handled in 

SimpleAgentJanitor Run.
LOG.error("SimpleAgentJanitor Failed.", t);

Is this what we want?

Comment (ankilosaurus) [reply deploy-service/teletraanservice/src/main/java/com/pinterest/teletraan/worker/SimpleAgentJanitor.java:62]: actually exception is caught in removeHost. but still better to catch uncaught. updated diff.

**Review by disparter — APPROVED** [review]

**Review by yongwen — APPROVED** [review]
